app/controllers/organisation_relationships_controller.rb

```diff
@@ -93,24 +93,37 @@ def create_managing_agent
     redirect_to managing_agents_organisation_path(related_organisation_id:)
   end
 
+  def remove_housing_provider
+    render "organisation_relationships/remove_housing_provider", layout: "application"
+  end
+
+  def delete_housing_provider
+    organisation_relationship_to_remove = OrganisationRelationship.find_by!(child_organisation_id: @organisation.id, parent_organisation_id: organisation_to_remove_id, relationship_type: OrganisationRelationship::OWNING)
+    delete(organisation_relationship_to_remove)
+    redirect_to housing_providers_organisation_path(removed_organisation_id: organisation_to_remove_id)
+  end
+
+  def delete(organisation_relationship_to_remove)
+    organisation_relationship_to_remove.destroy!
```

Comment:
I don't think you need to extract this :)

Reply:
True, I was copying the pattern we used in sharing content between housing provider/managing agent but I'll see if there's anything more substantial we can share in this method (otherwise I'll remove it)